### PR TITLE
Generic delete

### PR DESCRIPTION
**Pull request by saitejatokala**

### CODE REVIEW

File: li-android-sdk-core/src/main/java/lithium/community/android/sdk/manager/LiClientManager.java [hunk 1]
```diff
@@ -752,6 +757,31 @@ public static LiClient getGenericQueryParamsGetClient(LiClientRequestParams liCl
     }
 
     /**
+     * Generic DELETE Client
+     *
+     * @param liClientRequestParams {@link LiClientRequestParams.LiGenericDeleteQueryParamsClientRequestParams} {@link LiQueryRequestParams}
```
Comment (kunalshrivastava) [inline li-android-sdk-core/src/main/java/lithium/community/android/sdk/manager/LiClientManager.java]: Link LiQueryRequestParams inside LiCLientRequestParams.LiGenericDeleteQueryParamsClientRequestParams

File: li-android-sdk-core/src/main/java/lithium/community/android/sdk/manager/LiClientManager.java [hunk 2]
```diff
@@ -752,6 +757,31 @@ public static LiClient getGenericQueryParamsGetClient(LiClientRequestParams liCl
...
+     * @return LiClient {@link LiClient}
+     */
+    public static LiClient getGenericDeleteQueryParamsGetClient(LiClientRequestParams liClientRequestParams) throws LiRestResponseException {
+        liClientRequestParams.validate(Client.LI_GENERIC_DELETE_QUERY_PARAMS_CLIENT);
+        Map<String, String> queryRequestParams = ((LiClientRequestParams.LiGenericDeleteQueryParamsClientRequestParams) liClientRequestParams).getLiQueryRequestParams();
+        String id = ((LiClientRequestParams.LiGenericDeleteQueryParamsClientRequestParams) liClientRequestParams).getId();
+        String extraPathAfterId = ((LiClientRequestParams.LiGenericDeleteQueryParamsClientRequestParams) liClientRequestParams).getExtraPathAfterId();
```
Comment (kunalshrivastava) [inline li-android-sdk-core/src/main/java/lithium/community/android/sdk/manager/LiClientManager.java]: change name

File: li-android-sdk-core/src/main/java/lithium/community/android/sdk/manager/LiClientManager.java [hunk 3]
```diff
@@ -752,6 +757,31 @@ public static LiClient getGenericQueryParamsGetClient(LiClientRequestParams liCl
...
+        DeleteClient deleteClient = ((LiClientRequestParams.LiGenericDeleteQueryParamsClientRequestParams) liClientRequestParams).getDeleteClient();
```
Comment (kunalshrivastava) [inline li-android-sdk-core/src/main/java/lithium/community/android/sdk/manager/LiClientManager.java]: change name

File: li-android-sdk-core/src/main/java/lithium/community/android/sdk/manager/LiClientManager.java [hunk 4]
```diff
@@ -752,6 +757,31 @@ public static LiClient getGenericQueryParamsGetClient(LiClientRequestParams liCl
...
+        String path = String.format("/community/2.0/%s/%s/%s", LiSDKManager.getInstance().getTenant(), deleteClient.getValue(), id);
```
Comment (kunalshrivastava) [inline li-android-sdk-core/src/main/java/lithium/community/android/sdk/manager/LiClientManager.java]: use stringbuilder

File: li-android-sdk-core/src/main/java/lithium/community/android/sdk/manager/LiClientManager.java (hunk 1 above)
Comment (saitejatokala) [inline li-android-sdk-core/src/main/java/lithium/community/android/sdk/manager/LiClientManager.java]: Done

File: li-android-sdk-core/src/main/java/lithium/community/android/sdk/manager/LiClientManager.java (hunk 2 above)
Comment (saitejatokala) [inline li-android-sdk-core/src/main/java/lithium/community/android/sdk/manager/LiClientManager.java]: Done

File: li-android-sdk-core/src/main/java/lithium/community/android/sdk/manager/LiClientManager.java (hunk 4 above)
Comment (saitejatokala) [inline li-android-sdk-core/src/main/java/lithium/community/android/sdk/manager/LiClientManager.java]: Done

File: li-android-sdk-core/src/main/java/lithium/community/android/sdk/manager/LiClientManager.java [hunk 5]
```diff
@@ -752,6 +757,36 @@ public static LiClient getGenericQueryParamsGetClient(LiClientRequestParams liCl
     }
 
     /**
+     * Generic DELETE Client
+     *
+     * @param liClientRequestParams {@link LiClientRequestParams.LiGenericDeleteQueryParamsClientRequestParams}
+     * @return LiClient {@link LiClient}
+     */
+    public static LiClient getGenericDeleteQueryParamsGetClient(LiClientRequestParams liClientRequestParams) throws LiRestResponseException {
```
Comment (kunalshrivastava) [inline li-android-sdk-core/src/main/java/lithium/community/android/sdk/manager/LiClientManager.java]: change name. provider method should not contains params in its name. Also it is not a GET client, its a DELETE client so, name should be
getGenericDeleteQueryParamsGetClient -> getGenericQueryDeleteClient

Comment (saitejatokala) [inline li-android-sdk-core/src/main/java/lithium/community/android/sdk/manager/LiClientManager.java]: Thanks. Done

**Pull request merged**